lib/CleanConsoleReporter.ts

```diff
@@ -0,0 +1,93 @@
+/* global require module */
+
+import { Options, Rule} from "./types";
+import type {Config} from '@jest/types';
+import type {AggregatedResult, TestResult} from '@jest/test-result';
+
+import {DefaultReporter} from "@jest/reporters";
+import {LogEntry, LogType} from "@jest/console/build/types";
```

Comment:
reaching into `build` is not supported, if `tsc` supports [package exports](https://nodejs.org/api/packages.html#packages_exports) it'll actually fail from Jest 27.

I'd be happy to merge a PR in Jest that exports the types you need to build this reporter in TS

Reply:
Oh that is good to know! I'll add it to my list of things to do today. Thanks!

Reply:
@SimenB https://github.com/facebook/jest/pull/11017